src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -317,6 +335,254 @@ def on_time_step__prepare(self, event: Event):
         event : vivarium.framework.event.Event
 
         """
-        population = self.population_view.subview(["age"]).get(event.index, query="alive == 'alive'")
+        population = self.population_view.subview(["age"]).get(
+            event.index, query="alive == 'alive'"
+        )
         population["age"] += to_years(event.step_size)
         self.population_view.update(population)
+
+    def assign_gen_pop_guardians(self, pop: pd.DataFrame) -> pd.DataFrame:
+        """
+
+        Parameters
+        ----------
+        pop: State stable of simulants
+
+        Returns
+        -------
+        pd.Dataframe that will be pop with an additional guardians column containing the index for that simulant's
+          guardian..
+        """
+
+        # Initialize column
+        pop["guardian_1"] = data_values.UNKNOWN_GUARDIAN_IDX
+        pop["guardian_2"] = data_values.UNKNOWN_GUARDIAN_IDX
+        # Helper lists
+        non_relatives = ["Roommate", "Other nonrelative"]
+        children = ["Biological child", "Adopted child", "Foster child", "Stepchild"]
+        parents = ["Parent", "Parent-in-law"]
+        partners = [
+            "Opp-sex spouse",
+            "Opp-sex partner",
+            "Same-sex spouse",
+            "Same-sex partner",
+        ]
+
+        # Get household structure for population to vectorize choices
+        # Non-GQ population
+        gen_population = pop.loc[~pop["household_id"].isin(data_values.GQ_HOUSING_TYPE_MAP)]
+        household_structure = self.get_household_structure(gen_population)
+
+        # Children helper index groups
+        children_of_rp_idx = household_structure.loc[
+            household_structure["relation_to_household_head_x"].isin(children)
```

Comment:
i think im going to double down here on my comment within `self.get_household_structure` (sorry these will read out of order) -- it would be helpful if you rename "relation_to_household_head_x" to something like "childs_relation_to_household_head_x" (same for age_x and the two _y cols)